### DISCUSSION TITLE
Please wait. Loading...

### DISCUSSION THREAD

**Discussion by podworx** [rrirower/DP2SceneGraphWizard#32] [Q&A] [answered]
Hello, all.

I was instructed to place issues here vs. the Roku community.  Thanks for the direction.

When launching a channel after the conversion, the splash screen displays followed by "Please wait. Loading...".  The screen has the correct background and topper image.  But, it hangs.

I've tested it on a couple different channels.

I'm using the Smart TV plugin for WordPress to generate the JSON feed.  Been using the plugin for years.

Here's an example feed:

https://scottwhitneyacademy.com/wp-json/tv/roku/

Any thoughts?

**Comment by rrirower** [comment]
Are you using the latest version of the Wizard tool?

**Reply by podworx** [reply]
I believe so. The version on the About page is 1.0.0.0.

**Reply by rrirower** [reply]
Check the main.brs file.  At the top, it should show:

```
' BrightScript code version 1.0.0.0-alpha-6
' https://github.com/rrirower
```

If it does not, you have an old version.  Please 'watch' this repo for version notifications if you are not doing so.

**Reply by podworx** [reply]
Yep, it looks correct:

' BrightScript code version 1.0.0.0-alpha-6
' https://github.com/rrirower
' Constant values.

**Reply by rrirower** [reply, marked as answer]
Alpha-6 introduced support for MRSS (xml) feed files.  As such, the code needed a way to distinguish between the two feed types it supports (xml and json).  It relies on the fact that most URLs contain the feed extension as of its path.  Since your feed is missing that extension, the code does not know what to do with it.

**Reply by podworx** [reply]
Aha! That was the issue.  I copied the file to my computer, renamed it to include .json as an extension, uploaded to my S3 server and was able to point to it and make it work.

Here's my challenge. . .the Smart TV plugin doesn't have the .json extension in the feed.  And, for the life of me, I can't find a .json file with the channel feed anywhere within the file structure of the site.

Is there something I could change in your code to simply assume the feed is a json feed?

Thanks!

**Reply by rrirower** [reply]
No.  This would not work.  It would break the support of both MRSS and JSON feed files.  

FYI...you can use the Channel Feed Manager software to maintain your feed file and load it to S3.

**Comment by ferdiworks** [comment]
@podworx 
The feed links you sent do not open in the browser window and must be active to work on OTT channels
https://scottwhitneyacademy.com/wp-json/tv/roku/
https://pdwxroku.com/cfs/wp-json/tv/roku/

**Reply by ferdiworks** [reply]
@podworx The domain name doesn't respond at scottwhitneyacademy.com
Are you using the WP smart TV plugin? I think it was discontinued because of the Sunsetting with Roku!

**Reply by podworx** [reply]
Both of these feeds show correctly in my browser.  I am using the WP Smart TV plugin. It is what is generating the JSON feed.  I've attached an example feed (I think).
[cfs.json.txt](https://github.com/rrirower/DP2SceneGraphWizard/files/12703450/cfs.json.txt)
